File: solutions/Leetcode_462/leetcode_462.cc
#include <algorithm>
#include <vector>

using namespace std;

class Solution1
{
public:
    int minMoves2(vector<int> &nums)
    {
        sort(nums.begin(), nums.end());
        int median = nums[nums.size() / 2];
        int sum = 0;
        for (int num : nums)
            sum += abs(num - median);
        return sum;
    }
};


class Solution2
{
public:
    int minMoves2(vector<int> &nums)
    {
        sort(nums.begin(), nums.end());
        int sum = 0, n = nums.size();
        for (int i = 0; i < n / 2; i++)
            sum += nums[n-1-i] - nums[i];
        return sum;
    }
};

File: solutions/Leetcode_462/leetcode_462.py
from typing import List


class Solution1:
    def min_moves_2(self, nums: List[int]) -> int:
        nums.sort()
        median = nums[len(nums)//2]
        return sum(abs(x-median) for x in nums)


class Solution2:
    def min_moves_2(self, nums):
        nums.sort()
        return sum(nums[~i] - nums[i] for i in range(len(nums)//2))